Robust_Lab.qmd

```diff
@@ -41,85 +47,295 @@ Use the `stars` data in `robustbase`. This data looks at the relationship betwee
+      annotate(x = 9.24, y = 200, label = "Percentile CI", geom="text", angle=90, color="red") +
+      geom_vline(xintercept = upper_median_ci, color = "red") +
+      geom_vline(xintercept = lower_median_ci_bca, color = "blue") +
+      annotate(x=9.05, y=200, label = "BCA CI", geom="text", angle=90, color="blue") +
+      geom_vline(xintercept = upper_median_ci_bca, color = "blue") 
+
+
+    ```
+
 3.  You want to test whether the following paired samples are significantly different from one another: pre = \[22,25,17,24,16,29,20,23,19,20\], post = \[18,21,16,22,19,24,17,21,23,18\]. Often researchers would run a paired sampled t-test, but you are concerned the data does not follow a normal distribution.
 
     a.  Calculate the paired differences, that is post - pre, which will result in a vector of paired differences (pdiff0 = post - pre)
 
+    ```{r}
+
+    pre <- c(22,25,17,24,16,29,20,23,19,20)
+    post <- c(18,21,16,22,19,24,17,21,23,18)
+    pdiff <- post-pre
+
+    ```
+
     b\. Calculate the mean of the paired differences (Xpdiff0)
 
+    ```{r}
+    mean_pdiff <- mean(pdiff)
+    ```
+
     d\. Bootstrap b) with replacement (pdiff1) and plot the histogram with `ggplot2`.
 
+    ```{r}
+     #| message: false
+    set.seed(12)
+    paired_boot <- two.boot(pre, post, FUN=mean, R=1000)
+    paired_df <- as.data.frame(paired_boot$t)
+    ggplot() + geom_histogram(data=paired_df,aes(V1), bins=25) +
+      labs(x='Difference of Means')
+    ```
+
     e\. Calculate the 95% confidence intervals (BCa). What can you infer from this?
 
+    ```{r}
+    set.seed(12)
+    paired_boot_ci <- boot.ci(paired_boot, type="bca")
+    paired_boot_ci
+    ```
+
+    We can infer that there is no statistically significant difference between the means of the two sample, since the CI includes 0.
+
     f\. Plot bootstrap mean along with 95% CIs (with `ggplot2`). Use annotate to note what the vertical lines represent.
 
+    ```{r}
+      #| message: false
+    lower_paired_ci <- paired_boot_ci$bca[4]
+    upper_paired_ci <- paired_boot_ci$bca[5]
+
+    ggplot() + geom_histogram(data=paired_df,aes(V1), bins=20) +
+      labs(x='Difference of Means') +
+      geom_vline(xintercept = mean(paired_df$V1)) +
+      annotate(x=1.4, y=15, label="Mean", vjust=2, geom="text", angle=90) +
+      geom_vline(xintercept = lower_paired_ci, color = "red") +
+      annotate(x = 4.75, y = 100, label = "BCA CI", geom="text", angle=90, color="red") +
+      geom_vline(xintercept = upper_paired_ci, color = "red")
+    ```
+
 4.  Pepper Joe measured the length and heat of 85 chili peppers. He wants to know if smaller peppers are hotter than longer peppers.
 
     ```{r}
-    #read data.table to read in
-    chili<- read.delim("/Users/jasongeller/Documents/03-Robust_Methods/data/chillis.csv")
+    set.seed(12)
+    chili<- read.delim("https://raw.githubusercontent.com/jgeller112/psy504-advanced-stats/main/slides/03-Robust_Methods/data/chillis.csv")
+      
```

Comment:
I wanted you to use the boot function. What is being bootstrapped is the coefficients. 

boot.function <- function(data, indices) {
resampled_data = data[indices,]
model<-glm(HEAT~LENGTH, data=resampled_data)
return(coefficients(model))}

results <- boot(chili, boot.function, R=10000)

boot.ci(results, type="bca", R=10000)